tests/components/russound_rio/test_config_flow.py

```diff
@@ -0,0 +1,136 @@
+"""Test the Russound RIO config flow."""
+
+from unittest.mock import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components.russound_rio.const import DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.core import HomeAssistant
+
+from .const import MOCK_CONFIG, MODEL
+
+
+async def test_form(hass: HomeAssistant, mock_russound) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is data_entry_flow.FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    with (
+        patch(
+            "homeassistant.components.russound_rio.async_setup_entry",
+            return_value=True,
+        ) as mock_setup_entry,
+        patch(
+            "homeassistant.components.russound_rio.config_flow.Russound",
+            return_value=mock_russound,
+        ),
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            MOCK_CONFIG,
+        )
+
+    assert result2["type"] is data_entry_flow.FlowResultType.CREATE_ENTRY
+    assert result2["title"] == MODEL
+    assert result2["data"] == MOCK_CONFIG
+    assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_form_cannot_connect(hass: HomeAssistant) -> None:
+    """Test we handle cannot connect error."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    with patch(
+        "homeassistant.components.russound_rio.config_flow.Russound.connect",
+        side_effect=TimeoutError,
+    ):
```

Comment:
In this case you also have to include mock_russound as parameter. you can then do `mock_russound.connect.side_effect = TimeoutError` to achieve the same without `with patch`